### ISSUE TITLE
Message signing with multisig wallet

### ISSUE THREAD

**Issue by mandelbit** [sparrowwallet/sparrow#193]
Is message signing supported by multisig wallets in Sparrow at present?

In the `Sign/Verify Message` tool there seems to be no option to sign (`Sign` button is greyed out), only `Verify`. Is this a bug or unsupported feature?

**Comment by mandelbit** [comment]
It'd be a useful feature once standards for HW multisig signing are reached.

As you say, it doesn't look like [HWI](https://github.com/bitcoin-core/HWI) supports this yet. I wonder if efforts are being made to introduce support?

**Comment by mandelbit** [comment]
> It's unsupported. Unfortunately standards for message signing are fragmented for single sig, and non-existent (afaik) for multisig. Additionally, I doubt any hardware wallets support it, and they would typically be the cosigners in a multisig.

What about signing with  single sig HW wallets? I have the same greyed out `Sign` button for that.

**Comment by craigraw** [comment]
> What about signing with single sig HW wallets? I have the same greyed out Sign button for that.

That's because you will need to perform the message signing on the device itself.

**Comment by craigraw** [comment]
I should add - assuming this HWW is configured as airgapped.

**Comment by mandelbit** [comment]
On this particular occasion I was testing with a HW connected via USB. Funnily enough, I tried again and now the signing option is available so unfortunately I can't really help much with debugging what was wrong initially

**Comment by craigraw** [comment]
Ok, I'm going to close this off. If you encounter the issue again, please add another comment.

**Comment by mandelbit** [comment]
It's not clear to me why you have closed the issue. Message signing with cosigners in (HW) multisig is a legitimate and useful feature request.

**Comment by mandelbit** [comment]
> > It's unsupported. Unfortunately standards for message signing are fragmented for single sig, and non-existent (afaik) for multisig. Additionally, I doubt any hardware wallets support it, and they would typically be the cosigners in a multisig.
> 
> What about signing with single sig HW wallets? I have the same greyed out `Sign` button for that.

This is a separate, resolved issue that is unrelated to the subject of this issue.

**Comment by craigraw** [comment]
> It's not clear to me why you have closed the issue. Message signing with cosigners in (HW) multisig is a legitimate and useful feature request.

Fair enough. I've reopened it. I do suspect though that it is unlikely to be resolved anytime soon. I hope I'm wrong!

**Comment by mandelbit** [comment]
Hopefully won't be too long. I was quite surprised to discover that there doesn't even seem to be support in e.g. Electrum for multisig wallets w/o HW cosigners - I'd just assumed it'd be available as a standard already.

I opened [an issue ](https://github.com/bitcoin-core/HWI/issues/524) in the HWI repo. Maybe I'll take a stab at it at some point.

**Comment by rkbtc** [comment]
Hey guys, is there any update on this issue? I actually want to use this feature but its still unavailable!

**Comment by craigraw** [comment]
It's not just up to Sparrow - there are no standards for such a feature available, let alone implementations in hardware wallets. See my comment above.

**Comment by Pleb4Fun** [comment]
This missing feature was also not intuitive for me. I understand the lack of missing standards.
Maybe some "info" could be shown on hover or next to the disabled "Sign" Button?

So there is currently no way to prove to the outside world that a certain multi sig address is mine?
I can create and sign a transaction and just not broadcast it - which was enough for my usecase to just prove it to myself...

**Comment by craigraw** [comment]
> I can create and sign a transaction and just not broadcast it

Yes, this is the best way currently

> Maybe some "info" could be shown on hover or next to the disabled "Sign" Button?

Added validation for multisig addresses and a tooltip for the address field in cc8dd59d:
<img width="863" alt="Screenshot 2022-08-02 at 11 12 00" src="https://user-images.githubusercontent.com/862166/182340414-bb57acf4-28fa-4ab9-ae67-c7aa673a607a.png">

**Comment by Pleb4Fun** [comment]
love it - you are on fire! 🥳 

(As I see in your commit the messages seem to be "hardcoded" not using i18n. Any plans for translating the Wallet or don't you think this is needed as Bitcoiners around the world do speak English? I didn't even notice it till that moment as I'm used to English as an IT guy but I could imagine there are people out there needing/preferring their local language... 
Just saw that RaspiBlitz uses some nice Translation Tool for finding community help in translating the web frontend: https://mobile.twitter.com/_cstenglein_/status/1551650260942028800)
